enterprise/api/v1/views.py

```diff
@@ -188,3 +195,80 @@ class EnterpriseCustomerEntitlementViewSet(EnterpriseReadOnlyModelViewSet):
     )
     filter_fields = FIELDS
     ordering_fields = FIELDS
+
+
+class EnterpriseCatalogViewSet(viewsets.ViewSet):
+    """
+    API views for `enterprise customer catalogs` api endpoint.
+    """
+    serializer_class = serializers.EnterpriseCourseCatalogReadOnlySerializer
+    permission_classes = (permissions.IsAuthenticated,)
+    authentication_classes = (OAuth2Authentication, SessionAuthentication, BearerAuthentication, JwtAuthentication)
+    throttle_classes = (ServiceUserThrottle,)
+
+    def list(self, request):
+        """
+        DRF view to list all catalogs.
+
+        Arguments:
+            request (HttpRequest): Current request
+
+        Returns:
+            (Response): DRF response object containing course catalogs.
+        """
+        # fetch all course catalogs.
+        catalog_api = CourseCatalogApiClient(request.user)
+        catalogs = catalog_api.get_all_catalogs()
+        serializer = self.serializer_class(catalogs, many=True)
+        return Response(serializer.data)
+
+    def retrieve(self, request, pk=None):  # pylint: disable=invalid-name
+        """
+        DRF view to get catalog details.
+
+        Arguments:
+            request (HttpRequest): Current request
+            pk (int): Course catalog identifier
+
+        Returns:
+            (Response): DRF response object containing course catalogs.
+        """
+        # fetch course catalog for the given catalog id.
+        catalog_api = CourseCatalogApiClient(request.user)
+        catalog = catalog_api.get_catalog(pk)
+
+        if not catalog:
+            logger.error("Unable to fetch API response for given catalog from endpoint '/catalog/%s/'.", pk)
+            raise NotFound("The resource you are looking for does not exist.")
+
+        serializer = self.serializer_class(catalog)
+        return Response(serializer.data)
+
+    @detail_route()
+    def courses(self, request, pk=None):  # pylint: disable=invalid-name
```

Comment:
As above; if this is enterprise-facing, why not use the EnterpriseCustomer UUID?